### DISCUSSION TITLE
How do I fix these black squares appearing in images?

### DISCUSSION THREAD

**Discussion by huangfanglong** [lshqqytiger/stable-diffusion-webui-directml#31] [Q&A] [answered]
For example:

![image](https://user-images.githubusercontent.com/89382502/223252721-54f83872-31d4-41cc-8526-0a68709720ce.png)
![image](https://user-images.githubusercontent.com/89382502/223256520-ae534433-64da-43e3-b6d1-4caae06e20d7.png)

I have searched briefly on Google and some suggested it's a NSFW filter and I tried to remove them but none of the method seems to be applicable to this version. There is no txt2img.py in \scripts folder, the txt2img.py in \modules and \repositories\stable-diffusion-stability-ai\scripts don't have the lines required to adjust.

Someone said there is a box to tick for 'Filter NSFW content' in Settings on the Web UI but it doesn't seem to be in this version neither.

I have been getting blacked out images or images with black squares over random parts of the image like this recently so I can only assume it's a NSFW filter(?), however that doesn't make much sense to me since the AI has accidentally generated bare breasted images before multiple times.

I already have --full-precision and --no-half in my COMMANDLINE ARGS.

**Comment by Miraihi** [comment, marked as answer]
No NSFW filter here at all.
If you're using hirez fix that may be the memory leak problem. I can't use hirez fix at all for example.
Try `--medvram --precision full --no-half --no-half-vae --opt-split-attention-v1 --opt-sub-quad-attention --disable-nan-check ` arguments. Maybe any of these will help.

**Reply by huangfanglong** [reply]
My full Command Line Arg currently is:

--opt-sub-quad-attention --always-batch-cond-uncond --disable-nan-check --autolaunch --medvram --precision full --no-half --opt-split-attention-v1

I will try to add --no-half-vae since it seems like that might be what I needed after googling it, since I'm getting these black squares precisely on upscaling generations. Thank you.

**Reply by Miraihi** [reply]
Keep in mind that for now that may be an unsolvable problem for now. DirectML\Pytorch flaws waiting to be resolved. You may try [Ultimate SD upscale](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111) extension that works similarly to hirez fix.

**Reply by Miraihi** [reply]
> I will try to add --no-half-vae since it seems like that might be what I needed after googling it, since I'm getting these black squares precisely on upscaling generations. Thank you.

Wait, what upscaler are you using? SwinIR 4x produces the same black squares for me. Other upscalers don't. I'm pretty satisfied with ESRGAN for most types of images.

**Reply by huangfanglong** [reply]
> > I will try to add --no-half-vae since it seems like that might be what I needed after googling it, since I'm getting these black squares precisely on upscaling generations. Thank you.
> 
> Wait, what upscaler are you using? SwinIR 4x produces the same black squares for me. Other upscalers don't. I'm pretty satisfied with ESRGAN for most types of images.

I used Foolhardy's Remacri, adding that line seemed to have fixed the issue.

**Comment by lshqqytiger** [comment]
If black squares randomly (e.g. its size) appear, it can be a memory issue.

**Reply by huangfanglong** [reply]
Adding --no-half-vae to my commandline set seemed to have fixed it for me, I will update here if it occurs again.
